This patch fixes several bugs:

1. The configure script emitted nested messages when checking for
gettext in -lintl.

2. $INTLLIBS was never set.

3. dcgettext() was never found in -lintl.

4. The gettext programs were never linked with -lintl.


To apply this in the gettext sources:

# cd /usr/local/src/gettext-0.10.35
(or wherever you have extracted the source tree)
# patch -p1 </PATH/TO/gettext-0.10.35-2-intllibs.diff

With some creative thinking, you should be able to apply this to a
gettext binary package as well.  Use `locate' to find the files.


diff -Nurp gettext-0.10.35.kn.1/m4/ChangeLog gettext-0.10.35.kn.2/m4/ChangeLog
--- gettext-0.10.35.kn.1/m4/ChangeLog	Fri May  1 06:21:54 1998
+++ gettext-0.10.35.kn.2/m4/ChangeLog	Wed Sep 23 17:05:26 1998
@@ -1,3 +1,11 @@
+1998-09-23  Kalle Olavi Niemitalo  <tosi@stekt.oulu.fi>
+
+	* gettext.m4 (AM_WITH_NLS): Set gt_cv_func_gettext_libintl=no if
+	bindtextdomain() isn't found in -lintl.  Don't use AC_CACHE_CHECK
+	together with AC_CHECK_LIB because that has its own caching.  Set
+	INTLLIBS=-lintl if gettext() and bindtextdomain() are in -lintl.
+	Use $INTLLIBS when checking for dcgettext() and _nl_msg_cat_cntr.
+
 1998-04-30  Ulrich Drepper  <drepper@cygnus.com>
 
 	* gettext.m4: Add AC_OUTPUT_COMMANDS in AM_GNU_GETTEXT to rewrite
diff -Nurp gettext-0.10.35.kn.1/m4/gettext.m4 gettext-0.10.35.kn.2/m4/gettext.m4
--- gettext-0.10.35.kn.1/m4/gettext.m4	Fri May  1 05:45:20 1998
+++ gettext-0.10.35.kn.2/m4/gettext.m4	Wed Sep 23 16:56:26 1998
@@ -44,14 +44,16 @@ AC_DEFUN(AM_WITH_NLS,
 	    [AC_TRY_LINK([#include <libintl.h>], [return (int) gettext ("")],
 	       gt_cv_func_gettext_libc=yes, gt_cv_func_gettext_libc=no)])
 
+	   dnl TODO: rename gt_cv_func_gettext_libintl to
+	   dnl gt_tmp_func_gettext_libintl or something, as the value
+	   dnl in config.cache is never used.
+	   dnl (ac_cv_func_gettext_libintl takes care of the caching.)
 	   if test "$gt_cv_func_gettext_libc" != "yes"; then
 	     AC_CHECK_LIB(intl, bindtextdomain,
-	       [AC_CACHE_CHECK([for gettext in libintl],
-		 gt_cv_func_gettext_libintl,
-		 [AC_CHECK_LIB(intl, gettext,
-		  gt_cv_func_gettext_libintl=yes,
-		  gt_cv_func_gettext_libintl=no)],
-		 gt_cv_func_gettext_libintl=no)])
+	       [AC_CHECK_LIB(intl, gettext,
+		 gt_cv_func_gettext_libintl=yes INTLLIBS=-lintl,
+		 gt_cv_func_gettext_libintl=no)],
+	       gt_cv_func_gettext_libintl=no)
 	   fi
 
 	   if test "$gt_cv_func_gettext_libc" = "yes" \
@@ -60,6 +62,13 @@ AC_DEFUN(AM_WITH_NLS,
 	      AM_PATH_PROG_WITH_TEST(MSGFMT, msgfmt,
 		[test -z "`$ac_dir/$ac_word -h 2>&1 | grep 'dv '`"], no)dnl
 	      if test "$MSGFMT" != "no"; then
+		dnl Use $INTLLIBS when testing for dcgettext() and
+		dnl _nl_msg_cat_cntr.  Otherwise, they wouldn't be found
+		dnl if they were in -lintl.
+		gt_tmp_oldlibs="$LIBS"
+		LIBS="$INTLLIBS $LIBS"
+		dnl The macros here don't care of whether dcgettext()
+		dnl is defined but the program may.
 		AC_CHECK_FUNCS(dcgettext)
 		AC_PATH_PROG(GMSGFMT, gmsgfmt, $MSGFMT)
 		AM_PATH_PROG_WITH_TEST(XGETTEXT, xgettext,
@@ -71,6 +80,7 @@ AC_DEFUN(AM_WITH_NLS,
 		  [CATOBJEXT=.mo
 		   DATADIRNAME=lib])
 		INSTOBJEXT=.mo
+		LIBS="$gt_tmp_oldlibs"
 	      fi
 	    fi
 	])
diff -Nurp gettext-0.10.35.kn.1/src/ChangeLog gettext-0.10.35.kn.2/src/ChangeLog
--- gettext-0.10.35.kn.1/src/ChangeLog	Fri May  1 04:48:48 1998
+++ gettext-0.10.35.kn.2/src/ChangeLog	Wed Sep 23 17:05:26 1998
@@ -1,3 +1,8 @@
+1998-09-23  Kalle Olavi Niemitalo  <tosi@stekt.oulu.fi>
+
+	* Makefile.am (LDADD): Use $(INTLLIBS) rather than
+	../intl/libintl.$la.
+
 1998-04-30  Ulrich Drepper  <drepper@cygnus.com>
 
 	* dir-list.c: Update Peter Miller's mail address.
diff -Nurp gettext-0.10.35.kn.1/src/Makefile.am gettext-0.10.35.kn.2/src/Makefile.am
--- gettext-0.10.35.kn.1/src/Makefile.am	Wed Apr 29 17:00:35 1998
+++ gettext-0.10.35.kn.2/src/Makefile.am	Wed Sep 23 17:04:11 1998
@@ -33,7 +33,7 @@ l = @l@
 
 INCLUDES = -I$(srcdir) -I.. -I$(top_srcdir)/lib -I../intl -I$(top_srcdir)/intl
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
-LDADD = ../lib/libnlsut.a ../intl/libintl.$la
+LDADD = ../lib/libnlsut.a $(INTLLIBS)
 
 SED = sed
 YACC = @YACC@
